src/app/client/src/app/modules/core/components/data-driven-filter/data-driven-filter.component.ts

```diff
@@ -0,0 +1,206 @@
+import { ConfigService, ResourceService, Framework, ToasterService, ServerResponse, } from '@sunbird/shared';
+import { Component, OnInit, Input, Output, EventEmitter, ApplicationRef, ChangeDetectorRef } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { FrameworkService, FormService } from './../../services';
+import * as _ from 'lodash';
+import { CacheService } from 'ng2-cache-service';
+import { Observable } from 'rxjs/Observable';
+
+@Component({
+  selector: 'app-data-driven-filter',
+  templateUrl: './data-driven-filter.component.html',
+  styleUrls: ['./data-driven-filter.component.css']
+})
+export class DataDrivenFilterComponent implements OnInit {
+  @Input() routerVal: string;
+  @Input() inPageFilter: boolean;
```

Comment:
we dont need this. We will keep consistency. In all place, if we apply a filter. It should change the route. Parent component will provide filterEnv and routeUrl. In parent component, we need to subscribe to query param and fetch data.